Documentation/Concepts/event_handlers_and_filters.md

```diff
@@ -0,0 +1,95 @@
+
+The stream processors plays a central role in the Runtime, and without it the Runtime would be completely useless. They also enforce the most important rules of the Event Sourced Runtime; that an event in a stream is not processed twice (unless the stream is being replayed) and that no event in a stream is skipped while processing.
+
+#### Event Processor
+
+All stream processors consists of an event stream and an event processor that processes the events in that stream. This is a building block internal in the Runtime. They are similar to, but is not to be confused by, the "front facing" [event processors]({{< ref "#event-processors" >}}). These internal event processors are uniquely identified by an event processor id GUID. Since the streams are also uniquely identified by a stream id we can identify each stream processor by their (stream id, event processor id) pair.
+
+When the processing of an event is completed it returns a processing result back to the stream processor. This results contains information on whether or not the processing succeeded or not. If it did not succeed it will say how many times it has attempted to process that event, whether or not it should retry and how long it will wait until retrying.
+
+### Rules
+
+In order to maintain the [idempotency](https://en.wikipedia.org/wiki/Idempotence) of your event processors and the predictability of the Runtime we need to establish a couple of rules on streams. Some of the rules are directly related to the attributes of a stream.
+
+* The ordering of the events cannot change
+* Events can only be appended to the end of the stream
+* Events cannot be removed from the stream
+* A partitioned stream cannot be changed to be unpartitioned and vice versa
+
+The Runtime will enforce these rules. You're not allowed to do anything that breaks these rules.
+
+## Processors
+
+If you're a bit confused by now don't worry, you're not alone. The word "event processor" is used many places with slightly different semantics. Stream processors are internal constructs and the kinds of event processors we'll be talking about now, however, are strictly the things that the applications will be interacting with.
```

Comment:
So as the quip here says, it is a bit confusing.

My suggestion for reworking this whole page:

- split out the streams part into it's own page, there's well enough content there for it's own stuff
- On the page where you explain Event Handlers and FIlters you can say that they are what we call "event processors" and link to the 2 different types of event processors we have and leave it at that. if the reader wants to really learn about stream processors then they can read about them.
- Mention that event handlers is a combination of 2 processors (filter and then the processors) and that they create streams (once again linking to the streams page)
- Images would do wonders, maybe we can make some cool ones tomorrow